src/include/mca.h

```diff
@@ -544,6 +570,66 @@ void operator/=(Matrix<T> &a, const Number &number) {}
 template <class Number, class T, class>
 void operator/=(const Number &number, Matrix<T> &a) {}
 
+template <class T>
+void transpose(Matrix<T> &a) {
+    // single mode
+    Matrix<T> output(a.shape());
+    if (threadNum() == 0 || limit() > a.size()) {
+        transposeSingleThread(a, output, 0, a.size());
+        a = std::move(output);
+        return;
+    }
+    // threadCalculation and taskNum
+    auto res = threadCalculationTaskNum(a.size());
+
+    // the return value of every task, use this to make sure every task is finished
+    std::vector<std::future<void>> returnValue(res.second - 1);
+    // assign task for every sub-thread
+    for (size_t i = 0; i < res.second - 1; i++) {
+        returnValue[i] =
+            threadPool().addTask([&a, start = i * res.first, len = res.first, &output]() {
+                transposeSingleThread(a, output, start, len);
+            });
+    }
+    // let main thread calculate took
+    transposeSingleThread(
+        a, output, (res.second - 1) * res.first, (a.size() - (res.second - 1) * res.first));
+
+    // make sure all the sub threads are finished
+    for (auto &item : returnValue) { item.get(); }
+    a = std::move(output);
+}
+
+template <class T, class O>
+void transpose(const Matrix<T> &a, Matrix<O> &output) {
+    // single mode
```

Comment:
Add `assert(a.rows() == output.colomns())` and `assert(a.columns() == output.rows())`.